openfl/transport/grpc/director_server.py

```diff
@@ -206,22 +208,29 @@ async def StreamMetrics(self, request, context):  # NOQA:N802
         """Request to stream metrics from the aggregator to frontend."""
         if not self.validate_caller(request, context):
             return
-        # We should probably set a name to the aggregator and verify it here.
-        # Moreover, we may save the experiment name in plan.yaml and retrieve it
-        # during the aggregator initialization
+
         logger.info(f'Request StreamMetrics for {request.experiment_name} experiment has got!')
 
-        async for message in \
-                self.director.stream_metrics(request.experiment_name, request.header.sender):
-            yield message
+        while True:
+            metric_dict = self.director.get_next_metric(
+                experiment_name=request.experiment_name, caller=request.header.sender
+            )
+            if len(metric_dict) == 0:
+                await asyncio.sleep(5)
+                continue
+            if metric_dict is None:
+                break
```

Comment:
We should check that `metric_dict is None` before we are checking `len(metric_dict) == 0` because `len(None)` raise exception `TypeError: object of type 'NoneType' has no len()`

Reply:
Code was altered. Please check up changes.